tests/components/mqtt/test_device_tracker.py

```diff
@@ -429,6 +429,122 @@ async def test_setting_device_tracker_location_via_lat_lon_message(
     assert state.state == STATE_UNKNOWN
 
 
+async def test_setting_device_tracker_location_via_auto_zone_message(
+    hass, mqtt_mock_entry_no_yaml_config, caplog
+):
+    """Test the automatic inference of zones via MQTT."""
+    await mqtt_mock_entry_no_yaml_config()
+    async_fire_mqtt_message(
+        hass,
+        "homeassistant/device_tracker/bla/config",
+        "{ "
+        '"name": "test", '
+        '"state_topic": "test-topic", '
+        '"json_attributes_topic": "attributes-topic" '
+        "}",
+    )
+
+    hass.states.async_set(
+        "zone.school",
+        "zoning",
+        {
+            "latitude": 30.0,
+            "longitude": -100.0,
+            "radius": 100,
+            "friendly_name": "School",
+        },
+    )
+
+    await hass.async_block_till_done()
+
+    state = hass.states.get("device_tracker.test")
+    assert state.attributes["source_type"] == "gps"
+
+    assert state.state == STATE_UNKNOWN
+
+    hass.config.latitude = 32.87336
+    hass.config.longitude = -117.22743
+
+    # test auto zone and gps attributes
+    async_fire_mqtt_message(
+        hass,
+        "attributes-topic",
+        '{"latitude":32.87336,"longitude": -117.22743, "gps_accuracy":1.5}',
+    )
+    async_fire_mqtt_message(hass, "test-topic", "auto_zone")
+
+    state = hass.states.get("device_tracker.test")
+    assert state.attributes["latitude"] == 32.87336
+    assert state.attributes["longitude"] == -117.22743
+    assert state.attributes["gps_accuracy"] == 1.5
+    assert state.attributes["source_type"] == "gps"
+    assert state.state == STATE_HOME
+
+    # test manual state override
+    async_fire_mqtt_message(hass, "test-topic", "Work")
+
+    state = hass.states.get("device_tracker.test")
+    assert state.state == "Work"
+
+    # test auto_zone
+    async_fire_mqtt_message(hass, "test-topic", "auto_zone")
+
+    state = hass.states.get("device_tracker.test")
+    assert state.state == STATE_HOME
+
+    # test auto_zone inferring correct school area
+    async_fire_mqtt_message(
+        hass,
+        "attributes-topic",
+        '{"latitude":30.0,"longitude":-100.0,"gps_accuracy":1.5}',
+    )
+
+    state = hass.states.get("device_tracker.test")
+    assert state.state == "School"
+
+
+async def test_setting_device_tracker_location_via_abbr_auto_zone_message(
+    hass, mqtt_mock_entry_no_yaml_config, caplog
+):
+    """Test the setting of auto_zone via abbreviated names and custom payloads via MQTT."""
+    await mqtt_mock_entry_no_yaml_config()
+    async_fire_mqtt_message(
+        hass,
+        "homeassistant/device_tracker/bla/config",
+        "{ "
+        '"name": "test", '
+        '"state_topic": "test-topic", '
+        '"json_attributes_topic": "attributes-topic", '
+        '"pl_aut_zn": "auto" '
```

Comment:
Update the tests to use `CONF_PAYLOAD_RESET`. You do not need to use the abbreviation here. Using `payload_reset` is fine here.